sdks/java/harness/jmh/src/main/java/org/apache/beam/fn/harness/jmh/logging/BeamFnLoggingClientBenchmark.java

```diff
@@ -206,6 +207,23 @@ public void testLoggingWithCustomData(
     }
   }
 
+  @Benchmark
+  @Threads(16) // Use several threads since we expect contention during logging
+  public void testLoggingWithQuotaEvent(
+      ManyExpectedCallsLoggingClientAndService client, ManageExecutionState executionState)
+      throws Exception {
+    try (Closeable state =
+        executionState.executionStateTracker.enterState(executionState.simpleExecutionState)) {
+      try (AutoCloseable ac =
+          new QuotaEvent.Builder()
+              .withOperation("test")
+              .withFullResourceName("//test.googleapis.com/abc/123")
+              .create()) {
+        LOG.warn("log me");
```

Comment:
Just curious, how do we validate that the proper MDC is in place? Seems like we have benchmarks but it isn't clear if we test the functionality.

Reply:
I have manually verified that MDC in general (not QuotaEvent) works end-to-end on Dataflow. I plan on adding an integration test, once all the required Dataflow pieces have rolled out.